src/main/java/seedu/duke/Storage.java

```diff
@@ -1,14 +1,196 @@
+            storeFileValuesLocally(storedTransactions, input);
+            System.out.println("* duke.txt loaded successfully! *");
+
+        } catch (MoolahException e) {
+            //Catch any parsing errors and throw the default StorageInputCorruptedException
+            throw new StorageInputCorruptedException();
+        } catch (IOException e) {
+            throw new StorageWriteErrorException();
+        }
+        return storedTransactions;
     }
+
+    /**
+     * Stores values from duke.txt to the program by parsing each line in the file.
+     *
+     * @param storedTransactions The TransactionList to hold the stored values from the file.
+     * @param input              The input from duke.txt to be processed.
+     * @throws MoolahException WHen there are parsing errors, due to corrupted data.
+     */
+
+    private void storeFileValuesLocally(TransactionList storedTransactions, Scanner input) throws MoolahException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DateFormats.DATE_STORAGE_OUTPUT_PATTERN.toString());
+
+        Command command = null;
+        while (input.hasNext()) {
+            String line = input.nextLine();
+            String[] splits = line.split(DELIMITER);
+            if (splits.length != 5) {
+                throw new StorageInputCorruptedException();
+            }
+
+            String type = splits[0];
+            String category = splits[1];
+            String amountString = splits[2];
+            //Date has been formatted in duke.txt and must be synthesized into the correct string format before parsing.
+            try {
+                LocalDate date = LocalDate.parse(splits[3], formatter);
+                String dateString = synthesizeDateString(date);
+
+
+                String description = splits[4];
+
+                String parametersInput = "t/" + type + " c/" + category + " a/" + amountString + " d/" + dateString
+                        + " i/" + description;
+                command = getCommand("add", parametersInput);
+
+                ParameterParser.parse(command, parametersInput);
+
+                // amount would be converted into an integer before being used in the addition of transaction locally
+                int amount = Integer.parseInt(splits[2]);
+
+                switch (type) {
+                case "expense":
+                    storedTransactions.addExpenseDuringStorage(description, amount, category, date);
+                    break;
+                case "income":
+                    storedTransactions.addIncomeDuringStorage(description, amount, category, date);
+                    break;
+                default:
+                    throw new StorageInputCorruptedException();
+                }
+            } catch (DateTimeParseException e) {
+                // If the date format is incorrect, which is due to corrupted date information
+                throw new StorageInputCorruptedException();
+            }
+
+        }
```

Comment:
Arrowhead code, might want to extract into a few more methods next time when you have the time to do it.